code/mission/import/xwingmissionparse.cpp

```diff
@@ -623,6 +626,237 @@ void parse_xwi_flightgroup(mission *pm, const XWingMission *xwim, const XWMFligh
+
+		
+
+	// object position and orientation
+	// NOTE: Y and Z are swapped after all operartions are perfomed
+
+	float objectPosX = oj->object_x;
+	float objectPosY = oj->object_y;
+	float objectPosZ = oj->object_z;
+
+	int mine_dist = 400; // change this to change the distance between the mines
+	/** The minefield is a square 2d along two planes (a,b) centred on the given position **/
+	float offsetAxisA = 0 -(mine_dist / 2 * (number_of_objects - 1)); // (- the distance to centre the grid)
+	float offsetAxisB = 0 -(mine_dist / 2 * (number_of_objects - 1));
+		
+	matrix orient;
+	xwi_determine_object_orient(&orient, oj);
+	p_object pobj;
+
+	// Copy objects in Parse_objects to set for name checking below
+	// This only needs to be done fully once per object group then can be added to after each new object
+	SCP_set<char[32]> objectSet;
+	for (int n = 0; n < Parse_objects.size(); n++) {
+		objectSet.insert(Parse_objects[n].name);
+	}
+
+	// now being to configure each object in the group (mines multiple)
+
+	for (int a = 0; a < number_of_objects; a++) { // make an a-b 2d grid from the mines
+		offsetAxisA += (mine_dist * a); // add a new row to the grid
+		for (int b = 0; b < number_of_objects; b++) { // for each increment along the a plane, add mines along b plane
+			offsetAxisB += (mine_dist * b);  // for each new row populate the column
+			
+			/** Now convert the grid (a,b) to the relavenat formation ie. (x,y) or (z,y) etc **/
+			auto ojxyz = xwi_determine_mine_formation_position(oj, objectPosX, objectPosY, objectPosZ, offsetAxisA, offsetAxisB);
+			vm_vec_scale(&ojxyz, 1000); // units are in kilometers (after processing by xwinglib which handles the
+										// factor of 160), so scale them up
+
+			pobj.orient = orient;
+			pobj.pos = ojxyz;
+
+			// regular name, regular suffix
+
+			char base_name[NAME_LENGTH];
+			char suffix[NAME_LENGTH];
+			char *newArray = new char[std::strlen(base_name) + std::strlen(suffix) + 1];
+			strcpy_s(base_name, object_type);
+
+			end_string_at_first_hash_symbol(base_name);
+
+			// find lowest unique n for the suffix
+			int n = 1;
+			sprintf(suffix, NOX(" %d"), n);
+
+			// objects don't have designations so they are limited to the name from the object_type and suffix
+			// names will be hidden in game so to avoid iterating multiple times, allow for suffix 999 (+4)
+			int char_overflow = static_cast<int>(strlen(base_name) + 4) - (NAME_LENGTH - 1);
+			if (char_overflow > 0) {
+				base_name[strlen(base_name) - static_cast<size_t>(char_overflow)] = '\0';
+			}
+			char croppedName[NAME_LENGTH];
+			strcpy_s(croppedName, base_name); // save base_name incase suffix needs changed
+
+			strcat_s(croppedName, suffix);
+			// Now check the name against the objectSet of names
+			auto fin = objectSet.end();
+			while (objectSet.find(croppedName) != fin) {
+				n++;
+				sprintf(suffix, NOX(" %d"), n);
+				strcpy_s(croppedName, base_name);
+				strcat_s(croppedName, suffix);
+			}
```

Comment:
Extract all of this into a separate function.

What is `newArray`?  It's not being used, and shouldn't be needed, so take it out.

Instead of hard-coding the number 4, use `strlen(suffix)` as in the sample code.

You don't need a separate `croppedName` variable.  Use `pobj.name` itself.

Change `pobj.name` to title case as soon as it's created.  You want the name that is checked to be the same as the name that is stored in the set.